### PR TITLE
[JENKINS-46879] Check permissions against the item where the pipeline  is in

### PR DESCRIPTION
**Pull request by alvarolobato**
# Description

See [JENKINS-46879](https://issues.jenkins-ci.org/browse/JENKINS-46879).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by alvarolobato** [comment]
@reviewbybees

**Comment by michaelneale** [comment]
:bug: seems to be a test error: 

```
testOrganizationFolderFactory(io.jenkins.blueocean.rest.impl.pipeline.OrganizationFolderTest)  Time elapsed: 26.284 sec  <<< ERROR!
java.lang.NullPointerException
	at io.jenkins.blueocean.rest.impl.pipeline.ScmResourceImpl.checkPermission(ScmResourceImpl.java:68)
	at io.jenkins.blueocean.rest.impl.pipeline.ScmResourceImpl.<init>(ScmResourceImpl.java:33)
	at io.jenkins.blueocean.rest.impl.pipeline.ScmResourceImpl.<init>(ScmResourceImpl.java:26)
```

**Comment by michaelneale** [comment]
Ah, a NPE on a Nonnull element (more evidence that nonnull annotation stuff is often just junk - you really can't trust it can you). 

One way: 

```
        if (acl == null) acl = Jenkins.getInstance().getACL();
```
(but no idea of the intention here, and if local to test).

**Comment by alvarolobato** [comment]
@vivek @michaelneale fix pushed. Let wait for CI.

**Comment by michaelneale** [comment]
:bee: once green - I think @vivek is happy with this. Should be ok to get on release/1.2 as well 👍

**Comment by vivek** [comment]
@michaelneale yes, once green, will merge and port it to release/1.2 wait for release/1.2 branch go green then do the release.

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/ScmResourceImpl.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public Link getLink() {
     }
 
     private @Nonnull User checkPermission(){
-        ACL acl = Jenkins.getInstance().getACL();
+        ACL acl = item.getACL();
```
Comment (jtnord) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/ScmResourceImpl.java]: I'm not sure about any ACL test here.  The permission IIUC is controlled by the underlying SCM (e.g. github) using a user based ket/token.
Thus an admin user may be in chanrge of creating the multibranch pipelines - but if a user can push to github why not allow the user to use BO to do it even if they can not configure the job?

This is certainly better than it was so it gets an approval from me - but seems questionable in the bigger scheme of things.
/cc @vivek @michaelneale

Comment (vivek) [reply blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/ScmResourceImpl.java]: @jtnord This was added as extra check as editing a Jenkinsfile on existing job was considered job configuration and that user is operating in terms of existing job and editing pipeline was attached to configure permission. Main idea behind this was to reject edit of pipeline for users without configure permission. We can relax this permission if needed, but thats anyways is orthogonal to this PR.

Comment (michaelneale) [reply blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/ScmResourceImpl.java]: yeah its all a bit odd to me - but then again, a "root" user that can be not allowed to access a folder (which is what alvaro was catering for) kind of also does my head in, so I am not sure I really understand what is going on. 

As you can configure a lot with the Jenkinsfile, it does make sense to trust the SCM as the source of access truth...

**Pull request merged**